syft/source/excluding_file_resolver_test.go

```diff
@@ -0,0 +1,122 @@
+package source
+
+import (
+	"io"
+	"os"
+	"testing"
+
+	"github.com/anchore/stereoscope/pkg/file"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExcludingResolver(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		locations      []string
+		excludeFn      func(string, os.FileInfo) bool
+		expectedLength int
+	}{
+		{
+			name:      "keeps locations",
+			locations: []string{"a", "b", "c"},
+			excludeFn: func(s string, info os.FileInfo) bool {
+				return false
+			},
+			expectedLength: 3,
+		},
+		{
+			name:      "removes locations",
+			locations: []string{"d", "e", "f"},
+			excludeFn: func(s string, info os.FileInfo) bool {
+				return true
+			},
+			expectedLength: 0,
+		},
+		{
+			name:      "removes first match",
+			locations: []string{"g", "h", "i"},
+			excludeFn: func(s string, info os.FileInfo) bool {
+				return s == "g"
+			},
+			expectedLength: 2,
+		},
+		{
+			name:      "removes last match",
+			locations: []string{"j", "k", "l"},
+			excludeFn: func(s string, info os.FileInfo) bool {
+				return s == "l"
+			},
+			expectedLength: 2,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			resolver := &mockResolver{
+				locations: test.locations,
+			}
+			excludingResolver := NewExcludingResolver(resolver, test.excludeFn)
+
+			fc, _ := excludingResolver.FilesByPath()
+			assert.Equal(t, test.expectedLength, len(fc))
```

Comment:
(totally optional: there's an `assert.Len` function that makes this slightly cleaner)